tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/AadAuthorityTests.cs

```diff
@@ -318,6 +318,31 @@ public void CanonicalAuthorityInitTest()
             Assert.AreEqual(UriCustomPortTailSlash, authority.AuthorityInfo.CanonicalAuthority);
         }
 
+        [TestMethod]
+        public void TenantSpecificAuthorityInitTest()
```

Comment:
Will anything fail? Can you pls. ensure we can do the right thing when calling ARM resources? - I believe they have to have the double /

Reply:
Yes, this test will fail if the value that is passed into withAuthority does not match the authority on the application object after it is constructed. 

For the arm resources, if they use the 
`WithAuthority(Uri authorityUri, bool validateAuthority = true)` 
method, it will bypass this code altogether use whatever the user passed in as authorityUri. also, do you know if the double / will be at the end? (https://login.microsoftonline.com/tenant//) or is t supposed to be (https://login.microsoftonline.com//tenant/)?

Reply:
all your tests shows happy path. please add the test cases that doesn't end up in a happy scenario. Explicitly I would like an ARM test case that validates that arm works.

---
In reply to: [306521638](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1281#discussion_r306521638) [](ancestors = 306521638)

Reply:
coming to think of it I believe I thought of resources. Still would like to see test cases that ensures it's possible to set a path which includes extra / before and after tenant id. This was possible before, so it's possible this is being used somewhere

---
In reply to: [306529133](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1281#discussion_r306529133) [](ancestors = 306529133,306521638)

Reply:
Agreed with Henrik, we need a regression test for this.